tests/unit/python/foglamp/services/core/scheduler/test_scheduler.py

```diff
@@ -0,0 +1,841 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+import asyncio
+import datetime
+import os
+import time
+import uuid
+import pytest
+import json
+from unittest import mock
+from unittest.mock import patch
+from aiohttp.test_utils import make_mocked_request
+from aiohttp.streams import StreamReader
+from multidict import CIMultiDict
+
+from foglamp.services.core.scheduler.scheduler import Scheduler, _FOGLAMP_ROOT
+from foglamp.services.core.scheduler.entities import IntervalSchedule, Task, Schedule, TimedSchedule, ManualSchedule, \
+    StartUpSchedule
+from foglamp.services.core.scheduler.exceptions import ScheduleNotFoundError
+# from tests.unit.python.foglamp.services.common.storage_layer_fixture import Storage as Storage
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+@pytest.allure.feature("unit")
+@pytest.allure.story("scheduler")
+class TestScheduler:
+    @classmethod
+    def setup_class(cls):
+        cls.Scheduler = Scheduler(core_management_host=None, core_management_port=None)
+        cls.Scheduler._storage = Storage()
```

Comment:
A dedicated Storage class for testing is somewhat dangerous. I see its value in reusability. However, it shouldn't be overly complex. It should be the simplest structure needed for a specific test.